lib/Page.js

```diff
@@ -71,6 +81,44 @@ Page.prototype.collectIncludedFiles = function (dependencies) {
   });
 };
 
+/**
+ * Records the front matter of the current page into this.frontMatter
+ * Removes the front matter of the current page before it is rendered
+ * @param includedPage a page with its dependencies included
+ */
+Page.prototype.collectFrontMatter = function (includedPage) {
+  const $ = cheerio.load(includedPage);
+  const frontMatter = $('frontmatter');
+  const includedFrontMatter = $('frontmatter div');
+  if (frontMatter[0] !== undefined) {
+    let frontMatterData = '';
+    if (includedFrontMatter[0] !== undefined) {
+      frontMatterData = $(includedFrontMatter)[0].children[0].data;
+    } else {
+      frontMatterData = frontMatter[0].children[0].data;
+    }
+    const formattedMatter = `${FRONT_MATTER_FENCE}\n${frontMatterData}${FRONT_MATTER_FENCE}`;
+    // Parse front matter data
+    const parsedData = fm(formattedMatter);
+    parsedData.attributes.src = this.pageSrc;
+    parsedData.attributes.title = this.pageTitlePrefix + parsedData.attributes.title;
+    this.frontMatter = parsedData.attributes;
+    // Title specified in site.json will override title specified in front matter
+    if (this.title) {
+      this.title = this.pageTitlePrefix + this.title;
+    } else {
+      this.title = this.frontMatter.title;
+    }
```

Comment:
This doesn't seem to "override title specified in front matter" (it does "take priority").
Isn't `this.frontMatter` used by `Site` though?